runtime/src/main/java/org/corfudb/protocols/wireprotocol/MsgHandlingFilter.java

```diff
@@ -0,0 +1,219 @@
+                result.add(aTransformFunction.apply(corfuMsg));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * A convenience constructor for when MsgHandlingFilter is being constructed by a filter and
+     * a single transformation function on the message.
+     *
+     * @param filter a function indicating whether the MsgHandlingFilter should filter the
+     *               message by applying the provided transformFunction on the messages.
+     * @param transformFunction a function representing a transformation on the message which will
+     *                          transform a message given the filter evaluates to true for a
+     *                          message. Transforming message to null is an example representing
+     *                          a message being dropped. Existence of a transformFunction is core
+     *                          to existence of an instances for this class and cannot be Null.
+     */
+    public MsgHandlingFilter(@NonNull Function<CorfuMsg, Boolean> filter,
+                             @NonNull Function<CorfuMsg, CorfuMsg> transformFunction) {
+        this(filter, Arrays.asList(transformFunction));
+    }
+
+    /**
+     * Constructing a MsgHandlingFilter. Provided filter indicates whether the functions
+     * representing transformations must be applied on a message.
+     *
+     * @param filter a function indicating whether the MsgHandlingFilter should filter the
+     *               message by applying the provided transformFunctions on the messages.
+     * @param transformFunctions a list of functions representing a transformations on the message
+     *                          which will transform a message given the filter evaluates to true
+     *                          for a message. Transforming messages to null is an example representing
+     *                          a message being dropped. Existence of transformFunctions is core
+     *                          to existence of instances for the class and cannot be Null.
+     */
+    public MsgHandlingFilter(@NonNull Function<CorfuMsg, Boolean> filter,
+                             @NonNull List<Function<CorfuMsg, CorfuMsg>> transformFunctions) {
+        this.filter = filter;
+        this.transformFunctions = new ArrayList<>();
+        this.transformFunctions.addAll(transformFunctions);
+    }
+
+    /**
+     * A convenience method for creating a handling filter which checks all inbound messages and
+     * drops them if they are in {@param toBeDroppedTypeSet}
+     *
+     * @return a handling filter that drops messages with types provided to this method
+     */
+    public static MsgHandlingFilter msgDropFilterFor(
+            Set<CorfuMsgType> toBeDroppedTypeSet) {
+        Function<CorfuMsg, Boolean> filter =
+                corfuMsg -> toBeDroppedTypeSet.contains(corfuMsg.getMsgType());
+        Function<CorfuMsg, CorfuMsg> droppingProcess = corfuMsg -> null;
```

Comment:
Question: I see that we are transforming the corfuMsg to null in case it is to be dropped. Do we still send a null packet to the server? Or we drop it completely and nothing is delivered?